Software/Python/Examples/gps_reading.py

```diff
@@ -1,163 +1,163 @@
+                print ("")
+                print (self.raw_line)
+
+
+    def read(self):
+        '''
+        Attempts 50 times at most to get valid data from GPS
+        Returns as soon as valid data is found
+        If valid data is not found, then clean up data in GPS instance
+        '''
+        valid = False
+        for i in range(50):
+            time.sleep(0.5)
+            self.raw_line = self.ser.readline().strip()
+            if self.validate(self.raw_line):
+                valid = True
+                break;
+
+        if valid:
+            return self.gga
+        else:
+            self.clean_data()
+            return []
+
+    def validate(self,in_line):
+        '''
+        Runs regex validation on a GPGAA sentence.
+        Returns False if the sentence is mangled
+        Return True if everything is all right and sets internal
+        class members.
+        '''
+        if in_line == "":
+            return False
+        if in_line[:6] != "$GPGGA":
+            return False
+
+        self.gga = in_line.split(",")
+        debug (self.gga)
+
+        #Sometimes multiple GPS data packets come into the stream. Take the data only after the last '$GPGGA' is seen
+        try:
+            ind=self.gga.index('$GPGGA',5,len(self.gga))
+            self.gga=self.gga[ind:]
+        except ValueError:
+            pass
+
+        if len(self.gga) != 15:
+            debug ("Failed: wrong number of parameters ")
+            debug (self.gga)
+            return False
+
+        for i in range(len(self.validation)-1):
+            if len(self.gga[i]) == 0:
+                debug ("Failed: empty string %d"%i)
+                return False
+            test = self.validation[i].match(self.gga[i])
+            if test == False:
+                debug ("Failed: wrong format on parameter %d"%i)
+                return False
+            else:
+                debug("Passed %d"%i)
+
+        try:
+            self.timestamp = self.gga[1]
+            self.lat = float(self.gga[2])
+            self.NS = self.gga[3]
+            self.lon = float(self.gga[4])
+            self.EW = self.gga[5]
+            self.quality = int(self.gga[6])
+            self.satellites = int(self.gga[7])
+            self.altitude = float(self.gga[9])
+
+            self.latitude = self.lat // 100 + self.lat % 100 / 60
+            if self.NS == "S":
+                self.latitude = - self.latitude
+            self.longitude = self.lon // 100 + self.lon % 100 / 60
+            if self.EW == "W":
+                self.longitude = -self.longitude
+        except ValueError:
+            debug( "FAILED: invalid value")
+
+        return True
 
 if __name__ =="__main__":
     gps = GROVEGPS()
     print("Reading GPS sensor for location . . . ")
-    print("If you are not seeing coordinats appear, your sensor needs to be")
+    print("If you are not seeing coordinates appear, your sensor needs to be")
```

Comment:
I liked the way it was previously spelled.  🥇

Reply:
😝